File: python/tracing/superagent/02_redact.py
"""Trace a Superagent redaction operation."""

import asyncio
from pathlib import Path

from _shared import (
    configure_environment,
    create_respan,
    create_superagent_client,
    example_marker,
    finish_respan,
)
from respan import propagate_attributes, workflow

SCRIPT_NAME = Path(__file__).name


@workflow(name=SCRIPT_NAME)
async def run_redact(text: str) -> tuple[str, object]:
    config = configure_environment()
    client = create_superagent_client()

    result = await client.redact(
        input=text, model=config.model, entities=["EMAIL", "PHONE"]
    )

    return result.redacted, result.findings


async def main() -> None:
    respan = create_respan(SCRIPT_NAME)
    marker = example_marker()
    try:
        with propagate_attributes(
            trace_group_identifier=SCRIPT_NAME,
            custom_identifier=marker,
            customer_identifier="superagent-example-user",
            thread_identifier=f"{marker}-thread",
            metadata={
                "example": "superagent_redact",
                "script": SCRIPT_NAME,
                "run_id": marker,
                "example_run_id": marker,
            },
        ):
            redacted, findings = await run_redact(
                "Contact Ada at ada@example.com or 415-555-0100."
            )
    finally:
        finish_respan(respan)

    print("redacted:", redacted)
    print("findings:", findings)


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/tracing/superagent/02_redact.ts
import {
  configureEnvironment,
  createRespan,
  createSuperagentClient,
  logExampleResult,
  runWithExampleTrace,
} from "./_shared.js";

const workflowName = "TypeScript Superagent Redact Example";

export async function redactExample(): Promise<void> {
  const respan = await createRespan("typescript-superagent-redact-example");
  await respan.initialize();

  try {
    const result = await runWithExampleTrace(respan, workflowName, async () => {
      const config = configureEnvironment();
      const client = await createSuperagentClient();

      try {
        return await client.redact({
          input: "Contact Ada at ada@example.com or 415-555-0100 before launch.",
          model: config.model,
          entities: ["email addresses", "phone numbers"],
        });
      } catch (error) {
        return {
          redacted: null,
          findings: [],
          usage: null,
          error: error instanceof Error ? error.message : String(error),
        };
      }
    });

    logExampleResult(workflowName, {
      redacted: result.redacted,
      findings: result.findings,
      usage: result.usage,
    });
  } finally {
  }
}

await redactExample();
